src/pytorch_lightning/overrides/fairscale.py

```diff
@@ -11,27 +11,44 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from typing import Optional, Union
+
 import torch.nn as nn
 
 import pytorch_lightning as pl
-from pytorch_lightning.overrides.base import _LightningModuleWrapperBase, unwrap_lightning_module
-from pytorch_lightning.utilities import _IS_WINDOWS, _module_available
+from pytorch_lightning.overrides.base import (
+    _LightningModuleWrapperBase,
+    _LightningPrecisionModuleWrapperBase,
+    unwrap_lightning_module,
+)
+from pytorch_lightning.utilities import rank_zero_deprecation
+from pytorch_lightning.utilities.imports import _IS_WINDOWS, _module_available
 
 _FAIRSCALE_AVAILABLE = not _IS_WINDOWS and _module_available("fairscale.nn")
 
-if _FAIRSCALE_AVAILABLE:
+
+if _FAIRSCALE_AVAILABLE:  # pragma: no-cover
     from fairscale.nn.data_parallel.sharded_ddp import ShardedDataParallel
 
     class LightningShardedDataParallel(_LightningModuleWrapperBase):
-        # Just do this for later docstrings
-        pass
+        def __init__(
+            self,
+            forward_module: Optional[Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase]] = None,
+            pl_module: Optional[Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase]] = None,
+        ) -> None:
+            self._validate_init_arguments(pl_module, forward_module)
+            super().__init__(forward_module=(pl_module or forward_module))
 
     def unwrap_lightning_module_sharded(wrapped_model: nn.Module) -> "pl.LightningModule":
+        rank_zero_deprecation(
+            "The function `unwrap_lightning_module_sharded` is deprecated in v1.8 and will be removed in v1.10. Access"
+            " the `LightningModule` directly through the strategy attribute `Strategy.lightning_module`."
```

Comment:
```suggestion
            "The function `unwrap_lightning_module_sharded` is deprecated in v1.8.0 and will be removed in v1.10.0. Access"
            "the `LightningModule` directly through the strategy attribute `Strategy.lightning_module`."
```

Reply:
I think the whitespace is needed :eyes:

Reply:
@rohitgr7 why this. It is inconsistent in how we have done it so far.

Reply:
ah! I see. at some places we just don't add `z` of `x.y.z`, but I try to add the complete version. I think we should start adding a complete version since it is more explicit and tell users that the deprecation happened in `v1.8.0` and not `v.1.8.1` since recently we deprecated `LightningDeepSpeedModule` in `v1.7.1`.

Reply:
That is great. I do not mind either way and always value consistency over a specific choice. It is a lifestyle I guess. Or a byproduct of taking too many math classes at uni 🤣 

I updated the versions, hopefully everywhere.